Realign buttons questionpage

katloboda opened the pull request:
This PR realigns the buttons on the question page depending on the size of the screen. On large screens the generate question button, Qset Name, and all-mine-other button, are all on the same row. On smaller screens, the Qset Name moves above the two buttons.

Commit:
Fixing styling on the question page

Commit:
fixing merge conflicts

Commit:
Adding lines at end of files

elliottyates commented:
To test this locally:

```
git checkout -b katloboda-realign_buttons_questionpage master
git pull https://github.com/katloboda/askup.git realign_buttons_questionpage
```

katloboda commented:
Go to this URL to test on Heroku: https://mighty-citadel-8094.herokuapp.com

Commit:
Fixing the alignment of the all-mine-other button

Commit:
Fixed the view for small devices

Commit:
Merge branch 'master' into realign_buttons_questionpage

katloboda commented:
I fixed Ray's comments about the issues in styling for small screen views (the all-mine-other button would break up into two lines). I fixed this by placing the Generate Question and all-mine-other buttons on two different lines.

elliottyates commented on app/assets/stylesheets/questions.css.scss:
```diff
@@ -1,48 +1,56 @@
-// just above the question list, since it wraps onto its own .row
-#all-mine-filter-container-xs {
+// On small screens (bootstrap xs) show-small-screen and replace with show-large-screen otherwise
+#show-small-screen {
```
Can we keep this file alphabetical? (roughly) -- e.g. have all-mine-other (a-) entries at the top, move these s-related entries down, etc -- that makes it easier to look for things.

elliottyates commented on app/assets/stylesheets/questions.css.scss:
```diff
@@ -1,48 +1,56 @@
-    display: block;
+  
+  #all-mine-other-group {
+    float: none !important;
```
Why is this `float` `!important` stuff needed? maybe a comment here would help, if it is needed, for people like me who aren't real CSS-savvy.

elliottyates commented on app/assets/stylesheets/questions.css.scss:
```diff
@@ -1,48 +1,56 @@
+    width: 100%;
+
+    a {
+      display: block;
```
Why is this needed? Same thing, comment would be helpful.

elliottyates commented on app/assets/stylesheets/questions.css.scss:
```diff
@@ -1,48 +1,56 @@
 }
-// End of (generate-question-button-filter relocation at different screen sizes)
+
+#generate-question-offset {
```
I have a few preferences with respect to the ID names. Generally an ID is unique on a page and should be used to identify the element itself (semantically), vs. specifying behavior. Usually when behaviors are the important thing one can use a class tag -- these are not required to be unique (so they're reusable), and by convention a class added to an element clues the developer reading the source that the element has some customized behavior or styling (he/she can then go look in the CSS files to find out how it should behave).

All of the `*-offset` and `show-*` IDs here do look like they're unique, but the IDs aren't semantically useful (I don't know what to expect inside the `#show-small-screen` element -- if it were a `.show-small-screen` class, I would easily be able to ascertain the behavior though). A suggestion would be to split the difference here: 
- make the `show-*` into classes, not IDs (that way you could theoretically use them on other elements to hide/show, and don't need to change the semantic markup of the IDs at all)
- keep the `*-offset`s as IDs, but make the names more semantically helpful, like `#all-mine-other-container` or `#all-mine-other-col` or something. (The idea is that in this case the padding is so specific to that exact element that making it a class wouldn't be very helpful.)

elliottyates commented on app/views/qsets/_show_large_screen.html.erb:
```diff
@@ -0,0 +1,11 @@
+<div class = "row">
```
please remove spacing around `=`

elliottyates commented on app/views/qsets/_show_small_screen.html.erb:
```diff
@@ -0,0 +1,15 @@
+<div class ="row">
```
spacing

elliottyates commented on app/views/qsets/show.html.erb:
```diff
@@ -1,22 +1,15 @@
-  <%= render "qset_context" %>
-  <div id="generate-question-button-filter">
-    <%= render "generate_question" %>
+  <div id ="show-large-screen">
```
spacing

elliottyates commented on app/views/qsets/show.html.erb:
```diff
@@ -1,22 +1,15 @@
   </div>
-  <div id="all-mine-filter-container">
-    <%= render "all_mine_other_filter" %>
+  <div id ="show-small-screen">
```
spacing

elliottyates commented:
Looks good. Thanks for updating mighty-citadel, I was able to test it there. I changed the name of [Math21b](https://mighty-citadel-8094.herokuapp.com/qsets/3) so it has a much longer name, and it looks great at all screen sizes.

Commit:
Fixing PR comments

elliottyates commented on app/views/qsets/show.html.erb:
```diff
@@ -1,22 +1,15 @@
-  <div id="generate-question-button-filter">
-    <%= render "generate_question" %>
+  <div class="show-large-screen">
+    <%= render "show_large_screen" %>
```
the class in the outer div makes sense now, that's great. Why did the partial change to `show_large_screen` though? As a class, `.show-large-screen` clues the developer about the behavior of that element, but as a file/partial name `show_large_screen` isn't very helpful to indicate what's going to be rendered (or what's in the file). I'd switch the filenames back for greater readability.

elliottyates commented:
This looks great, and thanks for organizing the code so well and for updating mighty-citadel so it's easy to test. Once the filenames are switched back this is ready for a merge!

Commit:
Changing names of partials

elliottyates commented:
:+1:
